prov/rxm/src/rxm_cq.c

```diff
@@ -161,7 +161,8 @@ static int rxm_finish_recv(struct rxm_rx_buf *rx_buf, size_t done_len)
 		if (ret)
 			return ret;
 	} else {
-		if (rx_buf->recv_entry->flags & FI_COMPLETION) {
+		if (rx_buf->recv_entry->flags & FI_COMPLETION ||
+		    rx_buf->ep->rxm_info->mode & FI_BUFFERED_RECV) {
```

Comment:
Please ignore my previous comment. :) I think we're good here. recv_entry would be available only after user posts a discard or claim.